fixes.inc

```diff
@@ -9793,34 +9889,99 @@ native BAD_SetPlayerName(playerid, const name[]) = SetPlayerName;
 			}
 		#endif
 
+		// ==================================
+		//  BEGIN: GetVehicleComponentInSlot
+		// ==================================
+		#if FIX_GetVehicleComponentInSlot
+			new componentstring[100];
+			switch (FIXES_GetVehicleComponentType(componentid))
+			{
+				#if FIXES_Single
+					case CARMODTYPE_STEREO:
+					{
+						FIXES_gsStereo[vehicleid] = componentid;
+					}
+					case CARMODTYPE_FRONT_BUMPER:
+					{
+						FIXES_gsFrontBumper[vehicleid] = componentid;
+					}
+					case CARMODTYPE_REAR_BUMPER:
+					{
+						FIXES_gsRearBumper[vehicleid] = componentid;
+					}
+					case CARMODTYPE_FRONT_BULLBAR:
+					{
+						FIXES_gsFrontBullbar[vehicleid] = componentid;
+					}
+					case CARMODTYPE_REAR_BULLBAR:
+					{
+						FIXES_gsRearBullbar[vehicleid] = componentid;
+					}
+				#else 
+					case CARMODTYPE_STEREO:
+					{
+						valstr(componentstring, componentid);
+						setproperty(CARMODTYPE_STEREO + _FIXES_PROPERTY_OFFSET, "", vehicleid, componentstring);
+					}
+					case CARMODTYPE_FRONT_BUMPER:
+					{
+						valstr(componentstring, componentid);
+						setproperty(CARMODTYPE_FRONT_BUMPER + _FIXES_PROPERTY_OFFSET, "", vehicleid, componentstring);
+					}
+					case CARMODTYPE_REAR_BUMPER:
+					{
+						valstr(componentstring, componentid);
+						setproperty(CARMODTYPE_REAR_BUMPER + _FIXES_PROPERTY_OFFSET, "", vehicleid, componentstring);
+					}
+					case CARMODTYPE_FRONT_BULLBAR:
+					{
+						valstr(componentstring, componentid);
+						setproperty(CARMODTYPE_FRONT_BULLBAR + _FIXES_PROPERTY_OFFSET, "", vehicleid, componentstring);
+					}
+					case CARMODTYPE_REAR_BULLBAR:
+					{
+						valstr(componentstring, componentid);
+						setproperty(CARMODTYPE_REAR_BULLBAR + _FIXES_PROPERTY_OFFSET, "", vehicleid, componentstring);
+					}
+				#endif
+			}
+		#endif
+		// ================================
+		//  END: GetVehicleComponentInSlot
+		// ================================
+
 		// =====================
 		//  BEGIN: OnVehicleMod
 		// =====================
-		new
-			modelid = GetVehicleModel__(vehicleid);
-		if (_FIXES_IN_RANGE(modelid, 400, 611 + 1))
-		{
-			if (_FIXES_IN_RANGE(componentid, 1000, 1191 + 1))
+		#if FIX_OnVehicleMod
+			new
+				modelid = GetVehicleModel__(vehicleid);
+			if (_FIXES_IN_RANGE(modelid, 400, 611 + 1))
 			{
-				if (FIXES_gscVehicleMods[(modelid - 400) * 6 + (componentid - 1000 >>> 5)] & 1 << (componentid - 1000 & 0b00011111))
+				if (_FIXES_IN_RANGE(componentid, 1000, 1191 + 1))
 				{
-					return FIXES_OnVehicleMod(playerid, vehicleid, componentid);
+					if (FIXES_gscVehicleMods[(modelid - 400) * 6 + (componentid - 1000 >>> 5)] & 1 << (componentid - 1000 & 0b00011111))
```

Comment:
Does this code still work with the new mods?  I suspect the answer is yes, since the problem was that they were being applied but weren't readable, so they were obviously still in the list of valid mods.  But better to be sure.

Reply:
If I understand you well, I don't really get what do you mean by `new mods`, but I tested with 2-3 vehicles. Everything was good, including lowriders and these race cars modded in x-alien.